### PR TITLE
feat: Table supports hidden columns

### LINKED ISSUE #46930: Table Column 添加 hidden 属性
### What problem does this feature solve?

动态隐藏某列是非常常见的需求，通过 onCell 较为麻烦，colSpan = 0 并不像文档所描述的整列不渲染。使用数组的方法实现较为麻烦

### What does the proposed API look like?

type ColumnType  =  {
    hidden: boolean
    // ...
}

### PR DESCRIPTION
**Pull request by madocto**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close: #46930 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Table supports hidden columns      |
| 🇨🇳 Chinese |    Table 组件支持隐藏列       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
看下 CodeFactor 的问题。

**Comment by li-jia-nan** [comment]
这个需要提交到 feature 分支

File: components/table/index.en-US.md [hunk 1]
```diff
@@ -208,6 +209,7 @@ One of the Table `columns` prop for describing the table's columns, Column has t
 | sortIcon | Customized sort icon | (props: { sortOrder }) => ReactNode | - | 5.6.0 |
 | title | Title of this column | ReactNode \| ({ sortOrder, sortColumn, filters }) => ReactNode | - |  |
 | width | Width of this column ([width not working?](https://github.com/ant-design/ant-design/issues/13825#issuecomment-449889241)) | string \| number | - |  |
+| hidden | Hidden this column | boolean | false | 5.13.0 |
```
Comment (li-jia-nan) [inline components/table/index.en-US.md:212]: `5.13.0` 今天就要发了，这个到下个月了，改成 `5.14.0` 吧

File: components/table/index.zh-CN.md [hunk 2]
```diff
@@ -209,6 +210,7 @@ const columns = [
 | sortIcon | 自定义 sort 图标 | (props: { sortOrder }) => ReactNode | - | 5.6.0 |
 | title | 列头显示文字（函数用法 `3.10.0` 后支持） | ReactNode \| ({ sortOrder, sortColumn, filters }) => ReactNode | - |  |
 | width | 列宽度（[指定了也不生效？](https://github.com/ant-design/ant-design/issues/13825#issuecomment-449889241)） | string \| number | - |  |
+| hidden | 隐藏列 | boolean | false | 5.13.0 |
```
Comment (li-jia-nan) [inline components/table/index.zh-CN.md:213]:
```suggestion
| hidden | 隐藏列 | boolean | false | 5.14.0 |
```

**Comment by xrkffgg** [comment]
@madocto 这个 5.13 发，等你

**Comment by xrkffgg** [comment]
这个 5.13

**Comment by li-jia-nan** [comment]
sorry，我搞错了……我以为要放在下个版本

**Comment by li-jia-nan** [comment]
那就 reopen 一下吧

**Pull request closed without merging**